Kubernetes v1.33 Mid Cycle Sneak Peek Blog (#50111)
kubernetes/website: content/en/blog/_posts/2025-03-24-kubernetes-1.33-sneak-peek.md

Reviewer: If you can wrap the Markdown source, that'll help localization teams.

---
layout: blog
title: 'Kubernetes v1.33: Sneak peek'
date: 2025-03-24

> slug: kubernetes-v1-33-sneak-peek

Author: Thanks!

Note to myself to fix this up in the template file @rytswd

author: >
  Agustina Barbetta,
  Aakanksha Bhende,
  Udi Hofesh,
  Ryota Sawada,
  Sneha Yadav
---

As the release of Kubernetes v1.33 approaches, the project continues to evolve. Features may be deprecated, removed, or replaced to improve the overall health of the project. This blog post outlines some planned changes for the v1.33 release, which the release team believes you should be aware of to ensure the continued smooth operation of your Kubernetes environment and to keep you up-to-date with the latest developments.  The information below is based on the current status of the v1.33 release and is subject to change before the final release date.

> ### The Kubernetes API Removal and Deprecation process

Reviewer 2: ```suggestion
## The Kubernetes API Removal and Deprecation process
```

To keep consistency with the hierarchy is proably best to have a h2 heading

> The Kubernetes project has a well-documented [deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/) for features. This policy states that stable APIs may only be deprecated when a newer, stable version of that same API is available and that APIs have a minimum lifetime for each stability level. A deprecated API has been marked for removal in a future Kubernetes release, it will continue to function until removal (at least one year from the deprecation), but usage will result in a warning being displayed. Removed APIs are no longer available in the current version, at which point you must migrate to using the replacement.

Reviewer 3: We use site-relative links where feasible.

```suggestion
The Kubernetes project has a well-documented [deprecation policy](/docs/reference/using-api/deprecation-policy/) for features. This policy states that stable APIs may only be deprecated when a newer, stable version of that same API is available and that APIs have a minimum lifetime for each stability level. A deprecated API has been marked for removal in a future Kubernetes release, it will continue to function until removal (at least one year from the deprecation), but usage will result in a warning being displayed. Removed APIs are no longer available in the current version, at which point you must migrate to using the replacement.
```

* Generally available (GA) or stable API versions may be marked as deprecated but must not be removed within a major version of Kubernetes.

* Beta or pre-release API versions must be supported for 3 releases after the deprecation.

> * Alpha or experimental API versions may be removed in any release without prior deprecation notice; this process can become a withdrawal in cases where a different implementation for the same feature is already in place.

Reviewer 2: This sentence is OK by my side.

> * Alpha or experimental API versions may be removed in any release without prior deprecation notice; this process can become a withdrawal in cases where a different implementation for the same feature is already in place.

Reviewer 4: right. alright . thanks for your work and explanation.

> Whether an API is removed as a result of a feature graduating from beta to stable, or because that API simply did not succeed, all removals comply with this deprecation policy. Whenever an API is removed, migration options are communicated in the [deprecation guide](https://kubernetes.io/docs/reference/using-api/deprecation-guide/).

Reviewer 3: ```suggestion
Whether an API is removed as a result of a feature graduating from beta to stable, or because that API simply did not succeed, all removals comply with this deprecation policy. Whenever an API is removed, migration options are communicated in the [deprecation guide](/docs/reference/using-api/deprecation-guide/).
```

## A note about User namespace for Pods to be Enabled by Default

> One of the oldest open KEPs today is [KEP-127](https://kep.k8s.io/127), Pod security improvement by using Linux [User namespaces](/docs/concepts/workloads/pods/user-namespaces/) for Pods. This KEP was first opened in late 2016, and after multiple iterations, had its alpha release in v1.25, initial beta in v1.30 (where it was disabled by default), and now is set to be a part of v1.33, where it is available by default.

Reviewer 2: Here an unfamiliar reader could not know what keps are, so maybe adding a link to https://www.kubernetes.dev/resources/keps/ could help them understand what your are talking about

Author: Thank you for the input! Let me rework the order here -- I agree that it would read better when it talks about the deprecation process, and then some details about the upcoming deprecation(s).

I would probably rephrase the section "A note about ..." to be a bit more engaging to highlight one takeaway from the release as well.

Author: I have updated this with 8023e57..4cd6cb6 -- so the highlight (i.e. "A note about..." section) is now titled as "Editors' choice". The rest of sneak peek contents have the heading of "Other sneak peek of Kubernetes v1.33". The suffix of "of Kubernetes v1.33" may be redundant, and I might consider dropping that (as you mentioned in the other comment).

Author: As suggested by @sftim , I have updated the heading to be "Featured improvement of Kubernetes v1.33".

> One of the oldest open KEPs today is [#127](https://github.com/kubernetes/enhancements/issues/127), Pod security improvement by using Linux [User namespaces](https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/) for Pods. This KEP was first opened in late 2016, and after multiple iterations, had its alpha release in v1.25, initial beta in v1.30 (where it was disabled by default), and now is set to be a part of v1.33, where it is available by default.

Reviewer 3: ```suggestion
One of the oldest open KEPs today is [#127](https://kep.k8s.io/127), Pod security improvement by using Linux [User namespaces](/docs/concepts/workloads/pods/user-namespaces/) for Pods. This KEP was first opened in late 2016, and after multiple iterations, had its alpha release in v1.25, initial beta in v1.30 (where it was disabled by default), and now is set to be a part of v1.33, where it is available by default.
```

> This support will not impact existing Pods unless you manually specify `pod.spec.hostUsers` to opt in. As highlighted in the [v1.30 sneak peek blog](http://pod.spec.hostUsers), this is an important milestone in preventing and/or mitigating vulnerabilities.

Author: Thanks! This was actually coming from the template, I'll make sure to update it there as well!

## Deprecations and removals for Kubernetes v1.33

> ### Deprecate v1.Endpoints ([KEP-4974](https://kep.k8s.io/4974))

Reviewer 2: ```suggestion
### Deprecate v1.Endpoints 
```

Add the reference to the KEP in the paragraph instead of into the title.

Author: I'm a bit torn about this one -- it is true that you can check the paragraph, and that's the point of having these blogs, making it easier for readers of any level to understand upcoming changes. But for those readers with technical understanding, it would be useful to check out the KEPs to find more.

This is my personal take, but I think KEP is such a great asset Kubernetes community has, and want to make it as accessible as possible. I could take this out from the title, and perhaps put it at the bottom of each section, saying something like "If you want to find more about this, read this KEP" -- what do you think?

Reviewer: Try this:
```suggestion
## Deprecation of the stable Endpoints API
```

This is not really an enhancement, like the other things we're giving a sneak peek into.

Author: This has been updated with 9fe4dba8c7

> ### Deprecate v1.Endpoints ([KEP 4974](https://github.com/kubernetes/enhancements/issues/4974))

Reviewer 3: ```suggestion
### Deprecate v1.Endpoints ([KEP-4974](https://kep.k8s.io/4974))
```

> The [EndpointSlices](https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/) API has been stable since v1.21, which effectively replaced the original Endpoints API. The original Endpoints API was simple and straightforward, but also posed some challenges when scaling to large numbers of network endpoints. There have been new Service features only added to EndpointSlices API such as dual-stack networking, making the original Endpoints API ready for deprecation.

Reviewer 3: ```suggestion
The [EndpointSlices](/docs/concepts/services-networking/endpoint-slices/) API has been stable since v1.21, which effectively replaced the original Endpoints API. The original Endpoints API was simple and straightforward, but also posed some challenges when scaling to large numbers of network endpoints. There have been new Service features only added to EndpointSlices API such as dual-stack networking, making the original Endpoints API ready for deprecation.
```

> This deprecation only impacts only those who use the Endpoints API directly from workloads or scripts; these users should migrate to use EndpointSlices instead. You can find more about the deprecation implications and migration plans in a dedicated blog post [Endpoints formally deprecated in favor of EndpointSlices](TBC).

Reviewer 2: Are you planning to maintain this blog post or to have the "Endpoints formally deprecated in favor of EndpointSlices" before this piece is published? If not remove the reverence to a TBC blog post.

If the "Endpoints formally deprecated in favor of EndpointSlices" will be published before this one I think that the best option is:

```suggestion
This deprecation only impacts only those who use the Endpoints API directly from workloads or scripts; these users should migrate to use EndpointSlices instead. You can find more about the deprecation implications and migration plans in a [dedicated blog post](TBC).
```

Author: Yes, the idea is to have a dedicated blog post, before this one goes out. But that one is still in draft, and may be tight to get it released before the mid cycle blog goes out. I'll keep it as is for now, but will update according to your suggestion later (this PR is already on hold)

> This deprecation only impacts only those who use the Endpoints API directly from workloads or scripts; these users should migrate to use EndpointSlices instead. You can find more about the deprecation implications and migration plans in a dedicated blog post [Endpoints formally deprecated in favor of EndpointSlices](TBC).

Reviewer 5: [nit] 
```suggestion
This deprecation only impacts those who use the Endpoints API directly from workloads or scripts; these users should migrate to use EndpointSlices instead. You can find more about the deprecation implications and migration plans in a dedicated blog post [Endpoints formally deprecated in favor of EndpointSlices](TBC).
```

> ### Deprecate `status.nodeInfo.kubeProxyVersion` field ([KEP-4004](https://kep.k8s.io/4004))

Reviewer 2: ```suggestion
### Deprecate `status.nodeInfo.kubeProxyVersion` field 
```

Add the reference to the KEP in the paragraph instead of into the title.

Reviewer: I agree that this is the right thing. Links within headings don't work well.

I'd write:
```suggestion
### Removal of kube-proxy version information in node status
```

I'm afraid to point it out, but: the existing heading is almost misleading: people might think we're deprecating the field, not removing a deprecated field.

Author: This has been updated with 7a100b2e1d

> ### Deprecate `status.nodeInfo.kubeProxyVersion` field ([KEP 4004](https://github.com/kubernetes/enhancements/issues/4004))

Reviewer 3: ```suggestion
### Deprecate `status.nodeInfo.kubeProxyVersion` field ([KEP-4004](https://kep.k8s.io/4004))
```

> Following its deprecation in v1.31, as highlighted in the [release announcement](/blog/2024/07/19/kubernetes-1-31-upcoming-changes/#deprecation-of-status-nodeinfo-kubeproxyversion-field-for-nodes-kep-4004-https-github-com-kubernetes-enhancements-issues-4004), the `.status.nodeInfo.kubeProxyVersion` field will be removed in v1.33. This field was set by kubelet, but its value was not consistently accurate. As it has been disabled by default since v1.31, the v1.33 release will remove this field entirely.

Reviewer 2: ```suggestion
Following its deprecation in v1.31, as highlighted in the [release announcement](/blog/2024/07/19/kubernetes-1-31-upcoming-changes/#deprecation-of-status-nodeinfo-kubeproxyversion-field-for-nodes-kep-4004-https-github-com-kubernetes-enhancements-issues-4004), the `status.nodeInfo.kubeProxyVersion` field will be removed in v1.33. This field was set by kubelet, but its value was not consistently accurate. As it has been disabled by default since v1.31, the v1.33 release will remove this field entirely.
```

> ### Host network support for Windows pods ([KEP-3503](https://kep.k8s.io/3503))

Reviewer 2: ```suggestion
### Host network support for Windows pods 
```
Add the reference to the KEP in the paragraph instead of into the title.

Reviewer: ```suggestion
### Removal of host network support for Windows pods
```

Author: This has been updated with db248bccb1

> Following its deprecation in v1.31, as highlighted in the [release announcement](https://kubernetes.io/blog/2024/07/19/kubernetes-1-31-upcoming-changes/#deprecation-of-status-nodeinfo-kubeproxyversion-field-for-nodes-kep-4004-https-github-com-kubernetes-enhancements-issues-4004), the `.status.nodeInfo.kubeProxyVersion` field will be removed in v1.33. This field was set by kubelet, but its value was not consistently accurate. As it has been disabled by default since v1.31, the v1.33 release will remove this field entirely.

Reviewer 3: ```suggestion
Following its deprecation in v1.31, as highlighted in the [release announcement](/blog/2024/07/19/kubernetes-1-31-upcoming-changes/#deprecation-of-status-nodeinfo-kubeproxyversion-field-for-nodes-kep-4004-https-github-com-kubernetes-enhancements-issues-4004), the `.status.nodeInfo.kubeProxyVersion` field will be removed in v1.33. This field was set by kubelet, but its value was not consistently accurate. As it has been disabled by default since v1.31, the v1.33 release will remove this field entirely.
```

> One of the oldest open KEPs today is [KEP-127](https://kep.k8s.io/127), Pod security improvement by using Linux [User namespaces](/docs/concepts/workloads/pods/user-namespaces/) for Pods. This KEP was first opened in late 2016, and after multiple iterations, had its alpha release in v1.25, initial beta in v1.30 (where it was disabled by default), and now is set to be a part of v1.33, where it is available by default.

Reviewer: Let's not sound like the feature will 100% be enabled by default. That's very likely but not a commitment we're announcing.

> ### Host network support for Windows pods ([KEP 3503](https://github.com/kubernetes/enhancements/issues/3503))

Reviewer 3: ```suggestion
### Host network support for Windows pods ([KEP-3503](https://kep.k8s.io/3503))
```

> Windows Pod networking aimed to achieve feature parity with Linux and provide better cluster density by allowing containers to use Node’s networking namespace. The original implementation landed as alpha with v1.26, but as it faced unexpected containerd behaviours, and alternative solutions were available, it has been decided that the KEP will be withdrawn and the code removed in v1.33.

Reviewer 6: ```suggestion
Windows Pod networking aimed to achieve feature parity with Linux and provide better cluster density by allowing containers to use the Node’s networking namespace. The original implementation landed as alpha with v1.26, but as it faced unexpected containerd behaviours, and alternative solutions were available, it has been decided that the KEP will be withdrawn and the code removed in v1.33.
```

> This support will not impact existing Pods unless you manually specify `pod.spec.hostUsers` to opt in. As highlighted in the [v1.30 sneak peek blog](/blog/2024/03/12/kubernetes-1-30-upcoming-changes/), this is an important milestone in preventing and/or mitigating vulnerabilities.

Reviewer 7: ```suggestion
This support will not impact existing Pods unless you manually specify `pod.spec.hostUsers` to opt in. As highlighted in the [v1.30 sneak peek blog](/blog/2024/03/12/kubernetes-1-30-upcoming-changes/), this is an important milestone in mitigating vulnerabilities.
```

> ## Other sneak peek of Kubernetes v1.33

Reviewer: How about:
```suggestion
## Selected other Kubernetes v1.33 improvements
```

> ## Sneak peek of Kubernetes v1.33

Reviewer 2: ```suggestion
## Upcoming changes
```

Author: I think I'd keep this as "sneak peek", because, at this point, we don't yet know if these changes would actually land in the v1.33.
"Upcoming changes" may be a good one, but I'm wondering if it loses a bit of fun sense?

The following list of enhancements is likely to be included in the upcoming v1.33 release. This is not a commitment and the release content is subject to change.

> ### In-Place vertical Pod scalability with mutable PodSpec for resources ([KEP-1287](https://kep.k8s.io/1287))

Reviewer 2: ```suggestion
### In-Place vertical Pod scalability with mutable PodSpec for resources 
```
Add the reference to the KEP in the paragraph instead of into the title.

Reviewer: ```suggestion
### Improvements to in-place vertical scaling for Pods
```

Author: This has been updated with 9e4ca8cb35

> ### In-Place vertical Pod scalability with mutable PodSpec for Resources ([KEP 1287](https://github.com/kubernetes/enhancements/issues/1287))

Reviewer 3: ```suggestion
### In-Place vertical Pod scalability with mutable PodSpec for resources ([KEP-1287](https://kep.k8s.io/1287))
```

When provisioning a Pod, you can use various resources such as Deployment, StatefulSet, etc. Scalability requirements may need horizontal scaling by updating the Pod replica count, or vertical scaling by updating resources allocated to Pod’s container(s). Currently, As PodSpec’s Container Resources are immutable, updating Pod’s container resources results in Pod restarts. But what if we could dynamically update the resource configuration for Pods without restarting them?

> 

Reviewer 2: You can change like:

> Currently, container resource definitions in a Pod’s spec are immutable, meaning any updates to meet scalability requirements require a Pod restart. KEP-1287 introduces in-place Pod updates, enabling vertical scaling for stateful processes without downtime, seamless scale-down during low traffic and dynamic resource allocation such as provisioning extra resources at startup and reducing them afterward. Initially released as an alpha feature in v1.27, it is expected to reach beta in v1.33.

This way it is far more coincise and explain exacly the difference between how it works now and how it will be.

> 

Reviewer: Oops. Wrong place.

> When provisioning a Pod, you can use various resources such as Deployment, StatefulSet, etc. Scalability requirements may need horizontal scaling by updating the Pod replica count, or vertical scaling by updating resources allocated to Pod’s container(s). Currently, As PodSpec’s Container Resources are immutable, updating Pod’s container resources results in Pod restarts. But what if we could dynamically update the resource configuration for Pods without restarting them?

Reviewer 6: ```suggestion
When provisioning a Pod, you can use various resources such as Deployment, StatefulSet, etc. Scalability requirements may need horizontal scaling by updating the Pod replica count, or vertical scaling by updating resources allocated to a Pod’s container(s). Currently, As PodSpec’s Container Resources are immutable, updating a Pod’s container resources results in Pod restarts. But what if we could dynamically update the resource configuration for Pods without restarting them?
```

Author: I find the singularity / plurality with "a Pod's container(s)" is quite confusing (and the original wording is already more complex than I like). What do you think updating this to something like the following instead?

```suggestion
When provisioning a Pod, you can use various resources such as Deployment, StatefulSet, etc. Scalability requirements may need horizontal scaling by updating the Pod replica count, or vertical scaling by updating container resources allocated to the Pod. Currently, As PodSpec’s Container Resources are immutable, updating any of the Pod’s container resources results in Pod restarts. But what if we could dynamically update the resource configuration for Pods without restarting them?
```

Reviewer: Try not to write "PodSpec"; we prefer `spec` in backticks separate from Pod in UpperCamelCase.

PodSpec is mostly something you see either as part of the OpenAPI document or in the source code. People operating Kubernetes see `spec` and `Pod` within manifests and often wouldn't see `PodSpec` at all.

Reviewer 2: Your last suggestion looks good. Better `spec.containers.resources`.
If you want to make it readable is can suggest something like:

```suggestion
When provisioning a Pod, you can use various resources such as Deployment, StatefulSet, etc. Scalability requirements may need horizontal scaling by updating the Pod replica count, or vertical scaling by updating resources allocated to Pod’s container(s). Currently, since container resources defined in Pod's `spec` are immutable, updating any of them results in Pod restarts. But what if we could dynamically update the resource configuration for Pods without restarting them?
```

> The [KEP-1287](https://kep.k8s.io/1287) is precisely to allow such in-place Pod updates. It opens up various possibilities of vertical scale-up for stateful processes without any downtime, seamless scale-down when the process has only little traffic, and even allocating larger resources during the startup and eventually lowering once the initial setup is complete. This has been released as alpha in v1.27, and is expected to land as beta in v1.33.

Reviewer 6: ```suggestion
The [KEP-1287](https://kep.k8s.io/1287) is precisely to allow such in-place Pod updates. It opens up various possibilities of vertical scale-up for stateful processes without any downtime, seamless scale-down when the process has only a little traffic, and even allocating larger resources during startup and eventually lowering once the initial setup is complete. This was released as alpha in v1.27, and is expected to land as beta in v1.33.
```

> ### DRA’s ResourceClaim Device Status graduates to beta ([KEP-4817](https://kep.k8s.io/4817))

Reviewer 2: ```suggestion
### DRA’s ResourceClaim Device Status graduates to beta
```

Add the reference to the KEP in the paragraph instead of into the title.

Reviewer: Watch out for implying that the graduation is definitely going to happen. We don't make promises in the mid-cycle blog unless SIG Architecture would confirm the promise has been made.

Author: The KEP link has been moved out from the heading with e0f4df1b4f

> The KEP [#1287](https://github.com/kubernetes/enhancements/issues/1287) is precisely to allow such in-place Pod updates. It opens up various possibilities of vertical scale-up for stateful processes without any downtime, seamless scale-down when the process has only little traffic, and even allocating larger resources during the startup and eventually lowering once the initial setup is complete. This has been released as alpha in v1.27, and is expected to land as beta in v1.33.

Reviewer 3: ```suggestion
The KEP [#1287](https://kep.k8s.io/1287) is precisely to allow such in-place Pod updates. It opens up various possibilities of vertical scale-up for stateful processes without any downtime, seamless scale-down when the process has only little traffic, and even allocating larger resources during the startup and eventually lowering once the initial setup is complete. This has been released as alpha in v1.27, and is expected to land as beta in v1.33.
```

> The `Devices` field in `ResourceClaim.Status`, introduced in v1.32, graduates to beta in v1.33. This field allows drivers to report device status data, improving both observability and troubleshooting capabilities.

Reviewer: ```suggestion
The `devices` field in with ResourceClaim `status`, originally introduced in the v1.32 release, is likely to graduate to beta in v1.33. This field allows drivers to report device status data, improving both observability and troubleshooting capabilities.
```

Author: Thanks, I will slightly update your suggestion with the following:

```suggestion
The `devices` field in ResourceClaim `status`, originally introduced in the v1.32 release, is likely to graduate to beta in v1.33. This field allows drivers to report device status data, improving both observability and troubleshooting capabilities.
```

Author: The wording has been updated with the above snippet with f18e1a4d88

> ### DRA’s ResourceClaim Device Status graduates to beta ([KEP-4817](https://github.com/kubernetes/enhancements/issues/4817))

Reviewer 3: ```suggestion
### DRA’s ResourceClaim Device Status graduates to beta ([KEP-4817](https://kep.k8s.io/4817))
```

> For example, reporting the interface name, MAC address, and IP addresses of network interfaces in the status of a `ResourceClaim` can significantly help in configuring and managing network services, as well as in debugging network related issues.

Author: Thanks, we have certainly tried to align with the style guide, but clearly missed this one. I am assuming you are talking about the API object with UpperCamelCase, and we will track this more closely going forward. 🙏

> You can find more information in [Dynamic Resource Allocation: ResourceClaim Device Status](https://kubernetes.io/docs/concepts/scheduling-eviction/dynamic-resource-allocation/#resourceclaim-device-status).

Reviewer 3: ```suggestion
You can find more information in [Dynamic Resource Allocation: ResourceClaim Device Status](/docs/concepts/scheduling-eviction/dynamic-resource-allocation/#resourceclaim-device-status).
```

> ### Ordered Namespace Deletion ([KEP-5080](https://github.com/kubernetes/enhancements/issues/5080))

Reviewer 3: ```suggestion
### Ordered Namespace Deletion ([KEP-5080](https://kep.k8s.io/5080))
```

> This KEP introduces a more structured deletion process for Kubernetes namespace to ensure secure and deterministic resource removal. The current semi-random deletion order can create security gaps or unintended behaviour, such as Pods persisting after their associated NetworkPolicies are deleted. By enforcing a structured deletion sequence that respects logical and security dependencies, this approach ensures Pods are removed before other resources. The design improves Kubernetes’s security and reliability by mitigating risks associated with non-deterministic deletions.

Reviewer 6: ```suggestion
This KEP introduces a more structured deletion process for Kubernetes namespaces to ensure secure and deterministic resource removal. The current semi-random deletion order can create security gaps or unintended behaviour, such as Pods persisting after their associated NetworkPolicies are deleted. By enforcing a structured deletion sequence that respects logical and security dependencies, this approach ensures Pods are removed before other resources. The design improves Kubernetes’s security and reliability by mitigating risks associated with non-deterministic deletions.
```

> ### Enhancements to Kubernetes Job Management and Persistent Volume Policies ([KEP-3850](https://kep.k8s.io/3850), [KEP-3998](https://kep.k8s.io/3998), [KEP-2644](https://kep.k8s.io/2644))

Author: Let me drop the PVC update here, and instead focus on update on Jobs only. The two KEPs are very closely related as they are both about indexed jobs, and that would probably flow more naturally.

> ### Enhancements to Kubernetes Job Management and Persistent Volume Policies ([KEP-3850](https://kep.k8s.io/3850), [KEP-3998](https://kep.k8s.io/3998), [KEP-2644](https://kep.k8s.io/2644))

Author: Updated the wording with 198fb58317

> ### Enhancements to Kubernetes Job Management and Persistent Volume Policies ([KEP-3850](https://kep.k8s.io/3850), [KEP-3998](https://kep.k8s.io/3998), [KEP-2644](https://kep.k8s.io/2644))

Reviewer 2: ```suggestion
### Enhancements to Kubernetes Job Management and Persistent Volume Policies 
```

Add the reference to the KEP in the paragraph instead of into the title.

Author: Along with the wording update, I have moved the KEP links from the heading with d9da762e40

> ### Enhancements to Kubernetes Job Management and Persistent Volume Policies ([KEP 3850](https://github.com/kubernetes/enhancements/issues/3850), [KEP 3998](https://github.com/kubernetes/enhancements/issues/3998), [KEP 2644](https://github.com/kubernetes/enhancements/issues/2644))

Reviewer 3: ```suggestion
### Enhancements to Kubernetes Job Management and Persistent Volume Policies ([KEP-3850](https://kep.k8s.io/3850), [KEP-3998](https://kep.k8s.io/3998), [KEP-2644](https://kep.k8s.io/2644))
```

> These three KEPs are all graduating to GA to provide better reliability for job and storage handling. [KEP-3850](https://kep.k8s.io/3850) provides per-index backoff limits for indexed jobs, while [KEP-3998](https://kep.k8s.io/3998) extends Job API to define conditions for making an indexed job as successfully completed when not all indexes are succeeded. For storage, [KEP-2644](https://kep.k8s.io/2644) ensures the deletion order of PV-PVC pairs to provide reliable resource cleanup in external storage infrastructure.

Reviewer 4: ```suggestion
These three KEPs are all graduating to GA to provide better reliability for job and storage handling. [KEP-3850](https://kep.k8s.io/3850) provides per-index backoff limits for indexed jobs, while [KEP-3998](https://kep.k8s.io/3998) extends Job API to define conditions for making an indexed job as successfully completed when not all indexes are succeeded. For storage, [KEP-2644](https://kep.k8s.io/2644) establishes the deletion order for a PV-PVC pairs to ensure reliable resource cleanup in external storage infrastructure.
```

> These three KEPs are all graduating to GA to provide better reliability for job and storage handling. KEP [#3850](https://github.com/kubernetes/enhancements/issues/3850) provides per-index backoff limits for indexed jobs, while KEP [#3998](https://github.com/kubernetes/enhancements/issues/3998) extends Job API to define conditions for making an indexed job as successfully completed when not all indexes are succeeded. For storage, KEP [#2644](https://github.com/kubernetes/enhancements/issues/2644) ensures the deletion order of PV-PVC pairs to provide reliable resource cleanup in external storage infrastructure.

Reviewer 3: ```suggestion
These three KEPs are all graduating to GA to provide better reliability for job and storage handling. [KEP-3850](https://kep.k8s.io/3850) provides per-index backoff limits for indexed jobs, while [KEP-3998](https://kep.k8s.io/3998) extends Job API to define conditions for making an indexed job as successfully completed when not all indexes are succeeded. For storage, [KEP-2644](https://kep.k8s.io/2644) ensures the deletion order of PV-PVC pairs to provide reliable resource cleanup in external storage infrastructure.
```

## Want to know more?

New features and deprecations are also announced in the Kubernetes release notes. We will formally announce what's new in [Kubernetes v1.33](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.33.md) as part of the CHANGELOG for that release.

You can see the announcements of changes in the release notes for:

* [Kubernetes v1.32](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.32.md)

* [Kubernetes v1.31](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.31.md)

* [Kubernetes v1.30](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.30.md)

> * [Kubernetes v1.29](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.29.md)

Author: Given the [most recent three minor releases](https://kubernetes.io/releases/) are the only ones supported, should I drop v1.29 mention here?

Author: This has been removed with db790517fe

## Get involved

The simplest way to get involved with Kubernetes is by joining one of the many [Special Interest Groups](https://github.com/kubernetes/community/blob/master/sig-list.md) (SIGs) that align with your interests. Have something you’d like to broadcast to the Kubernetes community? Share your voice at our weekly [community meeting](https://github.com/kubernetes/community/tree/master/communication), and through the channels below. Thank you for your continued feedback and support.

> - Follow us on Bluesky [@Kubernetesio](https://bsky.app/profile/kubernetes.io) for the latest updates

Author: I'm not using Bluesky much, but I guess it should be `@kubernetes.io` as you say 👍

```suggestion
- Follow us on Bluesky [@kubernetes.io](https://bsky.app/profile/kubernetes.io) for the latest updates
```

> - Follow us on Bluesky [@Kubernetesio](https://bsky.app/profile/kubernetes.io) for the latest updates

Author: This has been updated with 412258522b

- Join the community discussion on [Discuss](https://discuss.kubernetes.io/)
- Join the community on [Slack](http://slack.k8s.io/)

> - Post questions (or answer questions) on [Stack Overflow](http://stackoverflow.com/questions/tagged/kubernetes)

Author: Oh good to know -- should we keep both? Or just Server Fault?

**Both**

```suggestion
- Post questions (or answer questions) on [Server Fault](https://serverfault.com/questions/tagged/kubernetes) or [Stack Overflow](http://stackoverflow.com/questions/tagged/kubernetes)
```

**Server Fault only**

```suggestion
- Post questions (or answer questions) on [Server Fault](https://serverfault.com/questions/tagged/kubernetes)
```

> - Post questions (or answer questions) on [Stack Overflow](http://stackoverflow.com/questions/tagged/kubernetes)

Reviewer 2: In this article (https://kubernetes.io/blog/2024/12/11/kubernetes-v1-32-release/) we keep Stack Overflow, so I think that having both is not a problem.
